libraries/AP_Scripting/AP_Scripting_CANSensor.cpp

```diff
@@ -79,4 +97,24 @@ void ScriptingCANBuffer::add_buffer(ScriptingCANBuffer* new_buff) {
     next->add_buffer(new_buff);
 }
 
+// Add a filter, will pass ID's that match value given the mask
+bool ScriptingCANBuffer::add_filter(uint32_t mask, uint32_t value) {
+
+    // Run out of filters
+    if (num_filters >= ARRAY_SIZE(filter)) {
+        return false;
+    }
+
+    // Add to list and increment
+    filter[num_filters].mask = mask;
+    filter[num_filters].value = value & mask;
+    num_filters++;
+    return true;
+}
+
+// Helper for DroneCAN msg ID
+bool ScriptingCANBuffer::add_DroneCAN_filter(uint16_t msg_id) {
+    return add_filter(0x00FFFF00, uint32_t(msg_id) << 8);
```

Comment:
I can see a strong argument that we don't need this helper, we could do it in lua with a example in the script. 

We could provide some others, setting a filter for only extend or none extended frames springs to mind.